lib/rocketfuel_api/service.rb

```diff
@@ -20,9 +20,11 @@ def name
   end
 
   def uri_suffix
-    file_path = RocketfuelApi.root.join('lib', 'config', 'endpoints_for_services.yaml')
-    endpoints = YAML.load_file(file_path)
-    endpoint  = endpoints['service'][name]
+    @endpoints ||= YAML.load_file(
```

Comment:
This caches the endpoints for one service instance, and not across multiple instances, but this would be harder to implement...

Reply:
Alternatively, we could have kept them in the classes. :wink:

Reply:
Hey, it seems it is a perfect situation for a `@@` variable indeed. I read so many times to avoid classes variable because they are shared between sublcasses, but that is exactly what we wanted here.

Reply:
Added it to master, thanks guigs :+1:

Reply:
I meant as methods as it was before, but... anyway. :smile:

Reply:
To clarify: I feel like the methods where as simple as it gets. They specified the behavior of the subclasses. Storing what the methods returned in a file and having to load that is already way more complex than it needs to be.

Reply:
Ah, I see. Well, you got a point. :smile: 

I thought of that because it was only different data stored in classes and not actually different behaviors...